x-pack/plugins/fleet/server/routes/setup/index.ts

```diff
@@ -35,6 +39,77 @@ export const FleetSetupResponseSchema = schema.object(
   }
 );
 
+async function createAgentEnrichment(
+  esClient: ElasticsearchClient,
+  soClient: SavedObjectsClientContract
+) {
+  // Sync agent policies in a new index that will be used for enrichment
+  // This could happen in the agent policy service when policies are created/updated
+  const agentPoliciesData = await agentPolicyService.list(soClient, {
+    perPage: 10000,
+  });
+
+  for (const agentPolicy of agentPoliciesData.items) {
+    await esClient.index({
+      index: 'fleet-agent-policy-enrich',
+      id: agentPolicy.id,
+      document: {
+        policy_id: agentPolicy.id,
+        is_managed: agentPolicy.is_managed,
+        data_output_id: agentPolicy.data_output_id,
+        monitoring_output_id: agentPolicy.monitoring_output_id,
+      },
+    });
+  }
+
+  // Create the enrich policy and pipeline, this should probably be part the ES fleet plugin if move forward with that
+  await esClient.enrich
+    .putPolicy({
+      name: 'fleet-agent-policy',
+      match: {
+        indices: 'fleet-agent-policy-enrich',
+        match_field: 'policy_id',
+        enrich_fields: ['is_managed', 'data_output_id', 'monitoring_output_id'],
```

Comment:
having `is_managed` will simplify a lot of fleet UI that have to currently do a policy lookup